lib/users.ts

```diff
@@ -14,8 +14,7 @@ export function getUserId(): string | undefined {
   // Try to get from `set-cookie` header
   const header = headers().get("set-cookie")
   if (header) {
-    // Note: This pattern can be improved after https://github.com/vercel/next.js/issues/47189 is resolved.
-    const match = header.match(/^id=([a-f0-9,-]*);/i)
+    const match = header.match(/id=([a-z0-9]+);/i)
```

Comment:
Here's the regex issue. The old regex worked before I gave the cookie an expiration data, which changes the structure of the set-cookie header.